Objects/typevarobject.c

```diff
@@ -1957,7 +1957,8 @@ typealias_reduce_impl(typealiasobject *self)
 static PyObject *
 typealias_subscript(PyObject *self, PyObject *args)
 {
-    if (((typealiasobject *)self)->type_params == NULL) {
+    typealiasobject *ta = ((typealiasobject *)self);
+    if (ta->type_params == NULL || PyTuple_GET_SIZE(ta->type_params) == 0) {
```

Comment:
What do you think of instead changing `typealias_new_impl` so it replaces an empty tuple with NULL? That has the advantage that we don't have two different possible representations for an equivalent type alias object.

Reply:
Good idea!